src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -594,7 +575,7 @@ public bool IsActivityPointer
 
         public IDataColumnMetadata Metadata { get; }
 
-        public DType AttachmentType => _attachmentType;
+        public DType AttachmentType => IsAttachment ? LazyTypeProvider.GetExpandedType(IsTable) : DType.Invalid;
```

Comment:
>AttachmentType => IsAttachmen [](http://example.com/codeflow?start=21&length=29)

Previously this returned null for non-attachment type. Please check at all places this property is used.

Reply:
yup, double checked. IMO, this is a better approach, DType.Invalid is more meaningful as a default than null

Reply:
100 % agree that this is better approach. Just wanted to call out the difference with previous behavior.